### DISCUSSION TITLE
Docker BASE_PATH configuration leads to "Static Media Failure"

### DISCUSSION THREAD

**Discussion by alucarddelta** [netbox-community/netbox#9830] [Q&A] [answered]
Hey,

I'm trying to use a Traefik to add a `BASE_PATH` mapping (traefik rule PathPrefix) to a fresh instance of Netbox via Docker. As per the instruction I added the `BASE_PATH=netbox/` parameter to the Netbox env file as well as the corresponding Traefik rules.

The docker compose builds correctly and Traefik reports the its config is good, however when I go to the path eg `http://hostname/netbox/` (Traefik mapping port 80 to 8080) in a browser I get a "Static Media Failure" when it trys to load the css.

The `BASE_PATH` config appears to be working, as when I use the API, I get successful requests using the additional path requirement. In saying that the docs css in the browser also does not load.

To see if it was my install (eg. missing css files), I ran the docker compose with the port mapping (8000:8080) and without the BASE_PATH config. This worked as expected.

Netbox version: `v3.2.5`
Netbox docker labels for Traefik:
```
    labels:
      - traefik.enable=true
      - traefik.http.routers.netbox.rule=PathPrefix(`/netbox/`)
      - traefik.http.routers.netbox.service=netbox
      - traefik.http.services.netbox.loadbalancer.server.port=8080
```
Added env to netbox.env: `BASE_PATH=netbox/
`

**Comment by candlerb** [comment, marked as answer]
The docker container uses nginx to serve the static content:
https://github.com/netbox-community/netbox-docker/blob/release/docker/nginx-unit.json

This serves under `/static/` and as far as I can see is not configurable.

The options for you I can see are:

* It might be possible to configure Traefik to proxy `/netbox/static/*` to `/static/*` on the target container, whilst still proxying all other `/netbox/*` to `/netbox/*`.  You'll have to check Traefik docs.
* You could try injecting a new version of `nginx-unit.json` at runtime using a volume mount
* Otherwise, you'll need to build your own container image using the original image as base, and replacing this file.
